pyproject.toml

```
@@ -1,5 +1,6 @@
 [build-system]
 requires = ["setuptools", "wheel"]
```

```suggestion
requires = ["setuptools>=40.6.0", "wheel"]
```

Per @gaborbernat, you actually are relying on a feature of a recent `setuptools` (only for the build isolation part - the non-isolated builds don't need to be pinned specially), so in your `build-system.requires`, you'll need a more recent version.

I'm still not sure why this is happening, but this might fix it.

---

looks like a [broad update](https://github.com/python-attrs/attrs/pull/499/commits/fc95d363d54f37c26fe2fb3fdf0048d03e633bd7) fixed it. let's try smaller now…

---

looks like we've got it! https://travis-ci.org/python-attrs/attrs/jobs/491005258

> I'm still not sure why this is happening, but this might fix it.

Wanna bet "caching"?

---

I don't know if it's technically "caching" -- IIUC Travis uses VM images with setuptools and other packages already installed. So, unless there's a specific version specification in the build/install chain, or unless a `pip install -U setuptools` or a `pip install --force-reinstall {pkg with setuptools install dependency}` is invoked, the stale version never gets bumped.

---

I think Paul's point was that the _isolated env_ should have installed the latest version.

---

yeah 😎 note isolation build are easily satisfy able, so no reason not to use a new builder version 😁in your requires part, when it's used is the same work to get version 30 as it's version latest